Documentation/anti-patterns.md

```diff
@@ -56,6 +57,17 @@ created and then immediately lost.  Implementing proper error handling for RPCs
 is crucial for maintaining the reliability and stability of your gRPC
 communication.
 
+### Difference between Dial and NewClient
```

Comment:
I would prefer changing the content under `### Dialing in gRPC` to talk about `NewClient` and swap the current content under the heading over here.